app/src/main/res/layout/dialog_add_note.xml

```diff
@@ -0,0 +1,68 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout
```

Comment:
I think using constraint layout should be preferred for all the layouts as they can become quite complex over time.

Reply:
Reason I picked LinearLayout as parent is because the note-keeper layout will probably stay a simple layout like it is. Complicating the main layout will lead to loss of note area space, so I don't think anyone would do that. Minor plus being it's also easier to read (child views rendered in order along an axis)

You think there are any foreseeable changes which would be better off with a ConstraintLayout?

Reply:
I agree that the layout is simple but it's better to make things future-proof.

Reply:
**Issue**: Use ConstraintLayout as root view instead of LinearLayout

Pros:
- Low effort - Will hardly take 10 mins
- Better performance in case someone decides to use a complex layout for the add note dialog

Cons:
- No immediate or foreseeable need (using a complicated layout will only lead to loss of screen area for the note content on an already small screen)
- Setting a precedent: Always use ConstraintLayout even if there's no need. (Also LinearLayout is more readable imo)

@mhutti1 can you weigh in on this?

Reply:
I would say always use ConstraintLayout, I see every other viewgroup as essentially deprecated.
Only in extreme scenarios where you are trying to squeeze every last bit of performance out would you use anything else and at that point you might try to use the new Compose library